types/src/write_set.rs

```diff
@@ -217,57 +183,60 @@ impl WriteOp {
         use WriteOp::*;
 
         match self {
-            Creation { data, .. } | Modification { data, .. } => data.len(),
-            Deletion { .. } => 0,
+            Creation(v) | Modification(v) => v.bytes().len(),
+            Deletion(..) => 0,
         }
     }
 
     pub fn metadata(&self) -> &StateValueMetadata {
         use WriteOp::*;
 
         match self {
-            Creation { metadata, .. } | Modification { metadata, .. } | Deletion { metadata } => {
-                metadata
-            },
+            Creation(v) | Modification(v) => v.metadata(),
+            Deletion(d_meta) => d_meta,
         }
     }
 
     pub fn get_metadata_mut(&mut self) -> &mut StateValueMetadata {
```

Comment:
nit: we probably can have `metadata()` and `get_metadata_mut()` consistent, either both with get or both without?